src/types.jl

```diff
@@ -196,7 +204,16 @@ isexpr(s::BasicSymbolic) = !issym(s)
 iscall(s::BasicSymbolic) = isexpr(s)
 
 @inline isa_SymType(T::Val{S}, x) where {S} = x isa BasicSymbolic ? Unityper.isa_type_fun(Val(SymbolicUtils.BasicSymbolic), T, x) : false
+
+"""
+    issym(x)
+
+Returns `true` if `x` is a `Sym`. If true, `nameof` must be defined
+on `x` and must return a `Symbol`.
+"""
+issym(x) = false
 issym(x::BasicSymbolic) = isa_SymType(Val(:Sym), x)
+
```

Comment:
```suggestion
issym(x) = isa_SymType(Val(:Sym), x)
```

Type checking is done in `isa_SymType`. 
https://github.com/JuliaSymbolics/SymbolicUtils.jl/blob/d009ca8e986b12982dce13794218ea6b632da617/src/types.jl#L198